crypto/evp_extra/p_ed25519_asn1.c

```diff
@@ -167,7 +179,19 @@ static int ed25519_priv_decode(EVP_PKEY *out, CBS *params, CBS *key) {
     return 0;
   }
 
-  return ed25519_set_priv_raw(out, CBS_data(&inner), CBS_len(&inner));
+  const uint8_t *public= NULL;
+  size_t public_len = 0;
+  if (pubkey) {
+    uint8_t padding;
+    if (!CBS_get_u8(pubkey, &padding) || padding != 0) {
```

Comment:
Are you referring to the padding byte? I left a discussion above on this as well.

https://luca.ntop.org/Teaching/Appunti/asn1.html has a pretty good summary guide on asn.1. 

Otherwise if you want the actual specification that covers the padding byte then [this](https://www.itu.int/rec/T-REC-X.680-X.693-202102-I/en) is what you are interested in. Specifically `x6901`.

Reply:
Ahh I see now. Thanks.

(Just an idea: could add a note about the encoding type, makes it more obvious why we suddenly look for a padding octet in the beginning).